commit 6498bf5800a302ef69e7f4914e727893f278bb2f
Author: Ramalingam C <ramalingam.c@intel.com>
Date:   Tue May 7 21:57:38 2019 +0530

    drm: revocation check at drm subsystem
    
    On every hdcp revocation check request SRM is read from fw file
    /lib/firmware/display_hdcp_srm.bin
    
    SRM table is parsed and stored at drm_hdcp.c, with functions exported
    for the services for revocation check from drivers (which
    implements the HDCP authentication)
    
    This patch handles the HDCP1.4 and 2.2 versions of SRM table.
    
    v2:
      moved the uAPI to request_firmware_direct() [Daniel]
    v3:
      kdoc added. [Daniel]
      srm_header unified and bit field definitions are removed. [Daniel]
      locking improved. [Daniel]
      vrl length violation is fixed. [Daniel]
    v4:
      s/__swab16/be16_to_cpu [Daniel]
      be24_to_cpu is done through a global func [Daniel]
      Unused variables are removed. [Daniel]
      unchecked return values are dropped from static funcs [Daniel]
    
    Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
    Acked-by: Satyeshwar Singh <satyeshwar.singh@intel.com>
    Reviewed-by: Daniel Vetter <daniel@ffwll.ch>
    Acked-by: Dave Airlie <airlied@gmail.com>
    Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Link: https://patchwork.freedesktop.org/patch/msgid/20190507162745.25600-5-ramalingam.c@intel.com

diff --git a/drivers/gpu/drm/drm_sysfs.c b/drivers/gpu/drm/drm_sysfs.c
index ecb7b33002bb..18b1ac442997 100644
--- a/drivers/gpu/drm/drm_sysfs.c
+++ b/drivers/gpu/drm/drm_sysfs.c
@@ -78,6 +78,7 @@ int drm_sysfs_init(void)
 	}
 
 	drm_class->devnode = drm_devnode;
+	drm_setup_hdcp_srm(drm_class);
 	return 0;
 }
 
@@ -90,6 +91,7 @@ void drm_sysfs_destroy(void)
 {
 	if (IS_ERR_OR_NULL(drm_class))
 		return;
+	drm_teardown_hdcp_srm(drm_class);
 	class_remove_file(drm_class, &class_attr_version.attr);
 	class_destroy(drm_class);
 	drm_class = NULL;